thunder/executors/data_dependent_partition.py

```diff
@@ -89,6 +89,7 @@ def __init__(self, trace: TraceCtx):
         # as it appears to be far off from being universal.
         # We use indices as hash values instead.
         bsym_id_to_node_map: list[int] = []
+        copy_nodes: list[Node] = []
```

Comment:
Would it be advantageous to use "fast to lookup" data structure here if we have a membership test below?

Reply:
Oh, right. I saw the in and thought if... Looking at PRs before coffee.